blog: remove authentication keyword from the flutter authentication and authorization article (#15720)
supabase/supabase: apps/www/_blog/2022-11-22-flutter-authorization-with-rls.mdx

---
title: 'Flutter Authorization with RLS'
description: Learn how you can secure your Flutter app using Supabase Row Level Security.
author: tyler_shukert
image: flutter-chat-auth/supabase-flutter-auth.jpeg
thumb: flutter-chat-auth/supabase-flutter-auth.jpeg
tags:
  - flutter
  - auth
date: '2022-11-22'
toc_depth: 3
---

This article is the second part of the Flutter tutorial series. During the series, you will learn how to build cross-platform apps without worrying about the backend.

In this article, I will show you how you can make a secure chat application by introducing authorization to the basic chat app that we created [previously](https://supabase.com/blog/flutter-tutorial-building-a-chat-app).

We will store the chat data on [Supabase](https://supabase.com/).
Supabase utilizes the built in authorization mechanism of PostgreSQL called [Row Level Security](https://supabase.com/docs/guides/auth/row-level-security) or RLS to prevent unauthorized access from accessing or writing data to your database.
RLS allows developers to define row-by-row conditions that evaluate to either `true` or `false` to either allow the access or deny it.

> We will take a look at more specific examples of authorization using RLS through out this article.

Reviewer: ```suggestion
We will take a look at more specific examples of authorization using RLS throughout this article.
```

### Step 4: Authorization with Row Level Security (RLS)

> At this point, we seemingly have a complete app, except if we open the app right now, we will see every user's room with all the messages that have ever been sent within the app by any user.

Reviewer: ```suggestion
At this point, we seemingly have a complete app. But if we open the app right now, we will see every users' room along with all the messages that have ever been sent.
```

> This is because we have not set up Row Level Security to prevent unauthorized users to access the rooms yet.

Reviewer: ```suggestion
This is because we have not set up Row Level Security to prevent users from accessing rooms that don't belong to them.
```

> There are two ways we can define Row Level Security policies in Supabase, with the GUI or through SQL, but we will set them using SQL today.

Reviewer: ```suggestion
There are two ways we can define Row Level Security policies in Supabase: with the GUI or through SQL. Today we will use SQL.
```
